In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Supermarket Sales EDA

## Importando e conhecendo o dataset

In [25]:
import pandas as pd
import numpy as np
import plotly.express as px

In [3]:
dfSales = pd.read_csv('/content/drive/MyDrive/supermarket-sales-eda/dados/supermarket_sales.csv', encoding='utf-8')
dfSales

Invoice ID Branch       City Customer type  Gender  \
0    750-67-8428      A     Yangon        Member  Female   
1    226-31-3081      C  Naypyitaw        Normal  Female   
2    631-41-3108      A     Yangon        Normal    Male   
3    123-19-1176      A     Yangon        Member    Male   
4    373-73-7910      A     Yangon        Normal    Male   
..           ...    ...        ...           ...     ...   
995  233-67-5758      C  Naypyitaw        Normal    Male   
996  303-96-2227      B   Mandalay        Normal  Female   
997  727-02-1313      A     Yangon        Member    Male   
998  347-56-2442      A     Yangon        Normal    Male   
999  849-09-3807      A     Yangon        Member  Female   

               Product line  Unit price  Quantity   Tax 5%      Total  \
0         Health and beauty       74.69         7  26.1415   548.9715   
1    Electronic accessories       15.28         5   3.8200    80.2200   
2        Home and lifestyle       46.33         7  16.2155   340.5255   
3         Health and beauty       58.22         8  23.2880   489.0480   
4         Sports and travel       86.31         7  30.2085   634.3785   
..                      ...         ...       ...      ...        ...   
995       Health and beauty       40.35         1   2.0175    42.3675   
996      Home and lifestyle       97.38        10  48.6900  1022.4900   
997      Food and beverages       31.84         1   1.5920    33.4320   
998      Home and lifestyle       65.82         1   3.2910    69.1110   
999     Fashion accessories       88.34         7  30.9190   649.2990   

          Date   Time      Payment    cogs  gross margin percentage  \
0     1/5/2019  13:08      Ewallet  522.83                 4.761905   
1     3/8/2019  10:29         Cash   76.40                 4.761905   
2     3/3/2019  13:23  Credit card  324.31                 4.761905   
3    1/27/2019  20:33      Ewallet  465.76                 4.761905   
4     2/8/2019  10:37      Ewallet  604.17                 4.761905   
..         ...    ...          ...     ...                      ...   
995  1/29/2019  13:46      Ewallet   40.35                 4.761905   
996   3/2/2019  17:16      Ewallet  973.80                 4.761905   
997   2/9/2019  13:22         Cash   31.84                 4.761905   
998  2/22/2019  15:33         Cash   65.82                 4.761905   
999  2/18/2019  13:28         Cash  618.38                 4.761905   

     gross income  Rating  
0         26.1415     9.1  
1          3.8200     9.6  
2         16.2155     7.4  
3         23.2880     8.4  
4         30.2085     5.3  
..            ...     ...  
995        2.0175     6.2  
996       48.6900     4.4  
997        1.5920     7.7  
998        3.2910     4.1  
999       30.9190     6.6  

[1000 rows x 17 columns]

In [4]:
dfSales.shape

(1000, 17)

In [5]:
dfSales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

In [6]:
dfSales.describe()

Unit price     Quantity       Tax 5%        Total        cogs  \
count  1000.000000  1000.000000  1000.000000  1000.000000  1000.00000   
mean     55.672130     5.510000    15.379369   322.966749   307.58738   
std      26.494628     2.923431    11.708825   245.885335   234.17651   
min      10.080000     1.000000     0.508500    10.678500    10.17000   
25%      32.875000     3.000000     5.924875   124.422375   118.49750   
50%      55.230000     5.000000    12.088000   253.848000   241.76000   
75%      77.935000     8.000000    22.445250   471.350250   448.90500   
max      99.960000    10.000000    49.650000  1042.650000   993.00000   

       gross margin percentage  gross income      Rating  
count              1000.000000   1000.000000  1000.00000  
mean                  4.761905     15.379369     6.97270  
std                   0.000000     11.708825     1.71858  
min                   4.761905      0.508500     4.00000  
25%                   4.761905      5.924875     5.50000  
50%                   4.761905     12.088000     7.00000  
75%                   4.761905     22.445250     8.50000  
max                   4.761905     49.650000    10.00000

In [7]:
dfSales['Branch'].unique()

array(['A', 'C', 'B'], dtype=object)

In [8]:
dfSales['City'].unique()

array(['Yangon', 'Naypyitaw', 'Mandalay'], dtype=object)

In [9]:
dfSales['Customer type'].unique()

array(['Member', 'Normal'], dtype=object)

In [10]:
dfSales['Payment'].unique()

array(['Ewallet', 'Cash', 'Credit card'], dtype=object)

In [11]:
dfSales['Product line'].unique()

array(['Health and beauty', 'Electronic accessories',
       'Home and lifestyle', 'Sports and travel', 'Food and beverages',
       'Fashion accessories'], dtype=object)

## Objetivos do estudo



*   Definir se a operação tem lucro real;
*   Estudar lucro e qtd. de vendas por linha produto;
*   Pontuação média do consumidor por unidade e no geral;
*   Gênero e método de pagamento.



## Tratamento de dados

Primeiro vamos remover a coluna Time, que não vai ter relevância no nosso estudo, e então converter o tipo de dado da coluna date de Object para DateTime e ordenar o dataframe pela data, do mais antigo para o mais recente.

In [12]:
dfSales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 17 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Time                     1000 non-null   object 
 12  Payment                  1000 non-null   object 
 13  cogs                     1000 non-null   float64
 14  gross margin percentage  

In [13]:
dfSales.drop('Time', inplace=True, axis=1)

In [14]:
dfSales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Invoice ID               1000 non-null   object 
 1   Branch                   1000 non-null   object 
 2   City                     1000 non-null   object 
 3   Customer type            1000 non-null   object 
 4   Gender                   1000 non-null   object 
 5   Product line             1000 non-null   object 
 6   Unit price               1000 non-null   float64
 7   Quantity                 1000 non-null   int64  
 8   Tax 5%                   1000 non-null   float64
 9   Total                    1000 non-null   float64
 10  Date                     1000 non-null   object 
 11  Payment                  1000 non-null   object 
 12  cogs                     1000 non-null   float64
 13  gross margin percentage  1000 non-null   float64
 14  gross income             

In [15]:
dfSales.head(3)

Invoice ID Branch       City Customer type  Gender  \
0  750-67-8428      A     Yangon        Member  Female   
1  226-31-3081      C  Naypyitaw        Normal  Female   
2  631-41-3108      A     Yangon        Normal    Male   

             Product line  Unit price  Quantity   Tax 5%     Total      Date  \
0       Health and beauty       74.69         7  26.1415  548.9715  1/5/2019   
1  Electronic accessories       15.28         5   3.8200   80.2200  3/8/2019   
2      Home and lifestyle       46.33         7  16.2155  340.5255  3/3/2019   

       Payment    cogs  gross margin percentage  gross income  Rating  
0      Ewallet  522.83                 4.761905       26.1415     9.1  
1         Cash   76.40                 4.761905        3.8200     9.6  
2  Credit card  324.31                 4.761905       16.2155     7.4

In [16]:
pd.to_datetime(dfSales['Date'], format='%m/%d/%Y')

0     2019-01-05
1     2019-03-08
2     2019-03-03
3     2019-01-27
4     2019-02-08
         ...    
995   2019-01-29
996   2019-03-02
997   2019-02-09
998   2019-02-22
999   2019-02-18
Name: Date, Length: 1000, dtype: datetime64[ns]

In [17]:
dfSales['Date'] = pd.to_datetime(dfSales['Date'], format='%m/%d/%Y')

In [18]:
dfSales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 16 columns):
 #   Column                   Non-Null Count  Dtype         
---  ------                   --------------  -----         
 0   Invoice ID               1000 non-null   object        
 1   Branch                   1000 non-null   object        
 2   City                     1000 non-null   object        
 3   Customer type            1000 non-null   object        
 4   Gender                   1000 non-null   object        
 5   Product line             1000 non-null   object        
 6   Unit price               1000 non-null   float64       
 7   Quantity                 1000 non-null   int64         
 8   Tax 5%                   1000 non-null   float64       
 9   Total                    1000 non-null   float64       
 10  Date                     1000 non-null   datetime64[ns]
 11  Payment                  1000 non-null   object        
 12  cogs                     1000 non-n

In [19]:
dfSales.sort_values('Date')

Invoice ID Branch       City Customer type  Gender  \
856  770-42-8960      B   Mandalay        Normal    Male   
567  651-88-7328      A     Yangon        Normal  Female   
245  530-90-9855      A     Yangon        Member    Male   
696  182-52-7000      A     Yangon        Member  Female   
523  133-14-7229      C  Naypyitaw        Normal    Male   
..           ...    ...        ...           ...     ...   
158  743-04-1105      B   Mandalay        Member    Male   
881  115-38-7388      C  Naypyitaw        Member  Female   
963  364-34-2972      C  Naypyitaw        Member    Male   
937  361-85-2571      A     Yangon        Normal  Female   
474  676-39-6028      A     Yangon        Member  Female   

               Product line  Unit price  Quantity  Tax 5%    Total       Date  \
856      Food and beverages       21.12         8   8.448  177.408 2019-01-01   
567     Fashion accessories       65.74         9  29.583  621.243 2019-01-01   
245      Home and lifestyle       47.59         8  19.036  399.756 2019-01-01   
696       Sports and travel       27.04         4   5.408  113.568 2019-01-01   
523       Health and beauty       62.87         2   6.287  132.027 2019-01-01   
..                      ...         ...       ...     ...      ...        ...   
158       Health and beauty       97.22         9  43.749  918.729 2019-03-30   
881     Fashion accessories       10.18         8   4.072   85.512 2019-03-30   
963  Electronic accessories       96.82         3  14.523  304.983 2019-03-30   
937       Sports and travel       89.48         5  22.370  469.770 2019-03-30   
474  Electronic accessories       64.44         5  16.110  338.310 2019-03-30   

         Payment    cogs  gross margin percentage  gross income  Rating  
856         Cash  168.96                 4.761905         8.448     6.3  
567         Cash  591.66                 4.761905        29.583     7.7  
245         Cash  380.72                 4.761905        19.036     5.7  
696      Ewallet  108.16                 4.761905         5.408     6.9  
523         Cash  125.74                 4.761905         6.287     5.0  
..           ...     ...                      ...           ...     ...  
158      Ewallet  874.98                 4.761905        43.749     6.0  
881  Credit card   81.44                 4.761905         4.072     9.5  
963         Cash  290.46                 4.761905        14.523     6.7  
937         Cash  447.40                 4.761905        22.370     7.4  
474         Cash  322.20                 4.761905        16.110     6.6  

[1000 rows x 16 columns]

A data na tabela está exibida no formato ano/mes/dia

In [20]:
dfSales = dfSales.sort_values('Date')

In [21]:
dfSales

Invoice ID Branch       City Customer type  Gender  \
856  770-42-8960      B   Mandalay        Normal    Male   
567  651-88-7328      A     Yangon        Normal  Female   
245  530-90-9855      A     Yangon        Member    Male   
696  182-52-7000      A     Yangon        Member  Female   
523  133-14-7229      C  Naypyitaw        Normal    Male   
..           ...    ...        ...           ...     ...   
158  743-04-1105      B   Mandalay        Member    Male   
881  115-38-7388      C  Naypyitaw        Member  Female   
963  364-34-2972      C  Naypyitaw        Member    Male   
937  361-85-2571      A     Yangon        Normal  Female   
474  676-39-6028      A     Yangon        Member  Female   

               Product line  Unit price  Quantity  Tax 5%    Total       Date  \
856      Food and beverages       21.12         8   8.448  177.408 2019-01-01   
567     Fashion accessories       65.74         9  29.583  621.243 2019-01-01   
245      Home and lifestyle       47.59         8  19.036  399.756 2019-01-01   
696       Sports and travel       27.04         4   5.408  113.568 2019-01-01   
523       Health and beauty       62.87         2   6.287  132.027 2019-01-01   
..                      ...         ...       ...     ...      ...        ...   
158       Health and beauty       97.22         9  43.749  918.729 2019-03-30   
881     Fashion accessories       10.18         8   4.072   85.512 2019-03-30   
963  Electronic accessories       96.82         3  14.523  304.983 2019-03-30   
937       Sports and travel       89.48         5  22.370  469.770 2019-03-30   
474  Electronic accessories       64.44         5  16.110  338.310 2019-03-30   

         Payment    cogs  gross margin percentage  gross income  Rating  
856         Cash  168.96                 4.761905         8.448     6.3  
567         Cash  591.66                 4.761905        29.583     7.7  
245         Cash  380.72                 4.761905        19.036     5.7  
696      Ewallet  108.16                 4.761905         5.408     6.9  
523         Cash  125.74                 4.761905         6.287     5.0  
..           ...     ...                      ...           ...     ...  
158      Ewallet  874.98                 4.761905        43.749     6.0  
881  Credit card   81.44                 4.761905         4.072     9.5  
963         Cash  290.46                 4.761905        14.523     6.7  
937         Cash  447.40                 4.761905        22.370     7.4  
474         Cash  322.20                 4.761905        16.110     6.6  

[1000 rows x 16 columns]

In [23]:
month = dfSales['Date'].dt.month
dfSales['Month'] = month
dfSales

Invoice ID Branch       City Customer type  Gender  \
856  770-42-8960      B   Mandalay        Normal    Male   
567  651-88-7328      A     Yangon        Normal  Female   
245  530-90-9855      A     Yangon        Member    Male   
696  182-52-7000      A     Yangon        Member  Female   
523  133-14-7229      C  Naypyitaw        Normal    Male   
..           ...    ...        ...           ...     ...   
158  743-04-1105      B   Mandalay        Member    Male   
881  115-38-7388      C  Naypyitaw        Member  Female   
963  364-34-2972      C  Naypyitaw        Member    Male   
937  361-85-2571      A     Yangon        Normal  Female   
474  676-39-6028      A     Yangon        Member  Female   

               Product line  Unit price  Quantity  Tax 5%    Total       Date  \
856      Food and beverages       21.12         8   8.448  177.408 2019-01-01   
567     Fashion accessories       65.74         9  29.583  621.243 2019-01-01   
245      Home and lifestyle       47.59         8  19.036  399.756 2019-01-01   
696       Sports and travel       27.04         4   5.408  113.568 2019-01-01   
523       Health and beauty       62.87         2   6.287  132.027 2019-01-01   
..                      ...         ...       ...     ...      ...        ...   
158       Health and beauty       97.22         9  43.749  918.729 2019-03-30   
881     Fashion accessories       10.18         8   4.072   85.512 2019-03-30   
963  Electronic accessories       96.82         3  14.523  304.983 2019-03-30   
937       Sports and travel       89.48         5  22.370  469.770 2019-03-30   
474  Electronic accessories       64.44         5  16.110  338.310 2019-03-30   

         Payment    cogs  gross margin percentage  gross income  Rating  Month  
856         Cash  168.96                 4.761905         8.448     6.3      1  
567         Cash  591.66                 4.761905        29.583     7.7      1  
245         Cash  380.72                 4.761905        19.036     5.7      1  
696      Ewallet  108.16                 4.761905         5.408     6.9      1  
523         Cash  125.74                 4.761905         6.287     5.0      1  
..           ...     ...                      ...           ...     ...    ...  
158      Ewallet  874.98                 4.761905        43.749     6.0      3  
881  Credit card   81.44                 4.761905         4.072     9.5      3  
963         Cash  290.46                 4.761905        14.523     6.7      3  
937         Cash  447.40                 4.761905        22.370     7.4      3  
474         Cash  322.20                 4.761905        16.110     6.6      3  

[1000 rows x 17 columns]

Pronto, agora que ordenamos por data e sabemos que o dataframe cobre o periodo de três meses, de jan/2019 até mar/2019, podemos começar a estudar ele mais a fundo.

## Análise exploratória

#### Lucro x Vendas

In [27]:
table = pd.pivot_table(dfSales, index='Month', values=['Invoice ID', 'gross income'], aggfunc={'Invoice ID':'count', 'gross income':'sum'})
table

Invoice ID  gross income
Month                          
1             352      5537.708
2             303      4629.494
3             345      5212.167

In [34]:
fig = px.line(table, x=table.index.get_level_values('Month'), y=table.columns, title='Pedidos x Lucro bruto')
fig.update_xaxes(title='Mês')
fig.update_yaxes(title='')
fig.show()

In [36]:
fig = px.histogram(table, barmode='group', x=table.index.get_level_values('Month'), y=table.columns, histfunc='sum', title='Crescimento lucro bruto x pedidos')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Mês')
fig.update_yaxes(title='')
fig.show()

In [37]:
fig = px.pie(table, names=table.index.get_level_values('Month'), values='gross income', title='Percentual de lucro bruto por mês')
fig.show()

#### a) Lucro e pedidos

In [38]:
dfSales.head()

Invoice ID Branch       City Customer type  Gender         Product line  \
856  770-42-8960      B   Mandalay        Normal    Male   Food and beverages   
567  651-88-7328      A     Yangon        Normal  Female  Fashion accessories   
245  530-90-9855      A     Yangon        Member    Male   Home and lifestyle   
696  182-52-7000      A     Yangon        Member  Female    Sports and travel   
523  133-14-7229      C  Naypyitaw        Normal    Male    Health and beauty   

     Unit price  Quantity  Tax 5%    Total       Date  Payment    cogs  \
856       21.12         8   8.448  177.408 2019-01-01     Cash  168.96   
567       65.74         9  29.583  621.243 2019-01-01     Cash  591.66   
245       47.59         8  19.036  399.756 2019-01-01     Cash  380.72   
696       27.04         4   5.408  113.568 2019-01-01  Ewallet  108.16   
523       62.87         2   6.287  132.027 2019-01-01     Cash  125.74   

     gross margin percentage  gross income  Rating  Month  
856                 4.761905         8.448     6.3      1  
567                 4.761905        29.583     7.7      1  
245                 4.761905        19.036     5.7      1  
696                 4.761905         5.408     6.9      1  
523                 4.761905         6.287     5.0      1

In [39]:
table = pd.pivot_table(dfSales, index=['City', 'Month'], values=['Invoice ID', 'gross income'], aggfunc={'Invoice ID':'count', 'gross income':'sum'})
table

Invoice ID  gross income
City      Month                          
Mandalay  1             111     1770.2885
          2             109     1639.2510
          3             112     1647.4925
Naypyitaw 1             122     1925.4610
          2             100     1568.3325
          3             106     1771.3830
Yangon    1             119     1841.9585
          2              94     1421.9105
          3             127     1793.2915

Todas as unidades são bem próximas em número de pedidos, vamos plotar um gráfico de barra para entender o crescimento, ou queda, de lucro bruto e vendas no periodo estudado

In [42]:
fig = px.histogram(table, x=table.index.get_level_values('Month'), y='gross income', color=table.index.get_level_values('City'), title='Crescimento do lucro bruto por cidade')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Mês')
fig.update_yaxes(title='Lucro bruto')
fig.show()

In [44]:
fig = px.histogram(table, x=table.index.get_level_values('Month'), y='Invoice ID', color=table.index.get_level_values('City'), title='Crescimento de pedidos por cidade')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Mês')
fig.update_yaxes(title='Qtd. pedidos')
fig.show()

#### b) Pontuação média por unidade

In [46]:
table = pd.pivot_table(dfSales, index='City', values='Rating', aggfunc='mean')
table

Rating
City               
Mandalay   6.818072
Naypyitaw  7.072866
Yangon     7.027059

Pontuação média: 6,97

#### c) Membro x Comum

In [54]:
dfSales['Customer type'].value_counts()

Member    501
Normal    499
Name: Customer type, dtype: int64

Temos 501 clientes membros para 499 clientes comuns. Relação de aproximadamente 1:1. Vamos ver se houve aumento de membros no decorrer do tempo.

In [58]:
fig = px.histogram(dfSales, x='Month', color='Customer type', title='Crescimento de clientes membros', barmode='group')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Mês')
fig.update_yaxes(title='Quantidade')
fig.show()

In [63]:
fig = px.histogram(dfSales, x='City', color='Customer type', title='Clientes membros x comuns por cidade', barmode='group')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Cidade')
fig.update_yaxes(title='Quatidade')
fig.show()

Levando em consideração a distribuição de membros por cidade e o crescimento de membros no periodo de três meses, podemos conculuir que a proporção é de 1:1, e que ou não é tão vantajoso ser membro ou não é tão divulgado como deveria.
O número de membros por unidade e mês se mantem praticamente o mesmo, com pouca variação.

#### d) Linha de produto

In [64]:
dfSales.head()

Invoice ID Branch       City Customer type  Gender         Product line  \
856  770-42-8960      B   Mandalay        Normal    Male   Food and beverages   
567  651-88-7328      A     Yangon        Normal  Female  Fashion accessories   
245  530-90-9855      A     Yangon        Member    Male   Home and lifestyle   
696  182-52-7000      A     Yangon        Member  Female    Sports and travel   
523  133-14-7229      C  Naypyitaw        Normal    Male    Health and beauty   

     Unit price  Quantity  Tax 5%    Total       Date  Payment    cogs  \
856       21.12         8   8.448  177.408 2019-01-01     Cash  168.96   
567       65.74         9  29.583  621.243 2019-01-01     Cash  591.66   
245       47.59         8  19.036  399.756 2019-01-01     Cash  380.72   
696       27.04         4   5.408  113.568 2019-01-01  Ewallet  108.16   
523       62.87         2   6.287  132.027 2019-01-01     Cash  125.74   

     gross margin percentage  gross income  Rating  Month  
856                 4.761905         8.448     6.3      1  
567                 4.761905        29.583     7.7      1  
245                 4.761905        19.036     5.7      1  
696                 4.761905         5.408     6.9      1  
523                 4.761905         6.287     5.0      1

In [69]:
table = pd.pivot_table(dfSales, index='Product line', values='Invoice ID', aggfunc='count')
table

Invoice ID
Product line                      
Electronic accessories         170
Fashion accessories            178
Food and beverages             174
Health and beauty              152
Home and lifestyle             160
Sports and travel              166

In [72]:
fig = px.pie(table, names=table.index.get_level_values('Product line'), values='Invoice ID', title='Percentual de pedido por linha de produto')
fig.show()

É bem distribuida a quantidade de pedidos, onde quase todos tem o mesmo percentual com relação ao total de pedidos. De uma visão geral, não tem uma linha que o público tem maior preferência. Vamos ver com relação ao lucro bruto.

In [73]:
table = pd.pivot_table(dfSales, index='Product line', values='gross income', aggfunc='sum')
table

gross income
Product line                        
Electronic accessories     2587.5015
Fashion accessories        2585.9950
Food and beverages         2673.5640
Health and beauty          2342.5590
Home and lifestyle         2564.8530
Sports and travel          2624.8965

Aparentemente, a proporção se mantém para o lucro bruto, onde todas as linhas tem um valor muito aproximado.

In [74]:
fig = px.pie(table, names=table.index.get_level_values('Product line'), values='gross income', title='Percentual de lucro por linha de produto')
fig.show()

Como comentado, a proporção se manteve com relação ao percentual, mas mudou na relação lucro/pedidos. Onde acessórios de moda que representava o maior número de pedidos teve o quarto maior lucro bruto. Sendo comidas e bebidas que obteve o maior lucro bruto, mas a diferença monetária entre os dois foi de apenas 100,00.

Vamos ver se existe então alguma relação com gênero.

In [77]:
fig = px.histogram(dfSales, x='Product line', y='Invoice ID', color='Gender', histfunc='count', title='Linha de produto x gênero (pedidos)', barmode='group')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Linha de produto')
fig.update_yaxes(title='Qtd. de pedidos')
fig.show()

Gênero feminino tem clara preferência aos produtos de esportes e viagem e acessórios de moda. Enquanto o gênero masculino tem preferência aos produtos de saúde e beleza.

Vamos ver se existe alguma preferência por unidade de mercado.

In [79]:
table = pd.pivot_table(dfSales, index=['City', 'Product line'], values='Invoice ID', aggfunc='count')
table

Invoice ID
City      Product line                      
Mandalay  Electronic accessories          55
          Fashion accessories             62
          Food and beverages              50
          Health and beauty               53
          Home and lifestyle              50
          Sports and travel               62
Naypyitaw Electronic accessories          55
          Fashion accessories             65
          Food and beverages              66
          Health and beauty               52
          Home and lifestyle              45
          Sports and travel               45
Yangon    Electronic accessories          60
          Fashion accessories             51
          Food and beverages              58
          Health and beauty               47
          Home and lifestyle              65
          Sports and travel               59

In [82]:
fig = px.histogram(table, x=table.index.get_level_values('Product line'), y='Invoice ID', color=table.index.get_level_values('City'), barmode='group', title='Linha de produto por cidade (pedidos)')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Linha de produto')
fig.update_yaxes(title='Qtd. de pedidos')
fig.show()

  Os principais produtos vendidos em Mandalay são de esportes e viagem e acessórios de moda.
  Em Naypyitaw são produtos alimenticios e acessórios de moda.
  Já em Yangon são casa e estilo de vida e eletrônicos.
  Os produtos de saúde e beleza são vendidas de mesma quantia em Mandalay e Naypyitaw, e em menor quantia em Yangon.

#### e) Método de pagamento

In [83]:
dfSales.head()

Invoice ID Branch       City Customer type  Gender         Product line  \
856  770-42-8960      B   Mandalay        Normal    Male   Food and beverages   
567  651-88-7328      A     Yangon        Normal  Female  Fashion accessories   
245  530-90-9855      A     Yangon        Member    Male   Home and lifestyle   
696  182-52-7000      A     Yangon        Member  Female    Sports and travel   
523  133-14-7229      C  Naypyitaw        Normal    Male    Health and beauty   

     Unit price  Quantity  Tax 5%    Total       Date  Payment    cogs  \
856       21.12         8   8.448  177.408 2019-01-01     Cash  168.96   
567       65.74         9  29.583  621.243 2019-01-01     Cash  591.66   
245       47.59         8  19.036  399.756 2019-01-01     Cash  380.72   
696       27.04         4   5.408  113.568 2019-01-01  Ewallet  108.16   
523       62.87         2   6.287  132.027 2019-01-01     Cash  125.74   

     gross margin percentage  gross income  Rating  Month  
856                 4.761905         8.448     6.3      1  
567                 4.761905        29.583     7.7      1  
245                 4.761905        19.036     5.7      1  
696                 4.761905         5.408     6.9      1  
523                 4.761905         6.287     5.0      1

In [91]:
fig = px.histogram(dfSales, x='Payment', title='Método de pagamento')
fig.update_xaxes(title='Método de pagamento')
fig.update_yaxes(title='Frequência')
fig.update_layout(bargap=0.5)
fig.show()

De forma geral, o método mais comum é e-wallet e dinheiro, mesma frequência, enquanto cartão crédito é menos frequente.

In [90]:
fig = px.histogram(dfSales, x='City', color='Payment', barmode='group', title='Método de pagamento preferido por cidade')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Cidade')
fig.update_yaxes(title='Frequência')
fig.show()

Em Yangon a preferência é claramente e-wallet, em Naypyitaw dinheiro e em Mandalay não há método de pagamento preferido.

In [93]:
fig = px.histogram(dfSales, x='Payment', color='Gender', barmode='group', title='Método de pagamento preferido por gênero')
fig.update_layout(bargap=0.5)
fig.update_xaxes(title='Método')
fig.update_yaxes(title='Frequência')
fig.show()

Método preferido pelo gênero feminino foi dinheiro, enquanto masculo e-wallet. Por curiosidade, o pagamento por cartão de crédito é preferido pelo gênero femino.

## Conclusões

*   O maior lucro foi no mês de janeiro, lucro de 5537.71, com aproximadamente 352 pedidos.
*   Houve uma queda no lucro de 5.9% comparando janeiro e março.
*   A cidade com o maior lucro é Naypyitaw, aproximadamente 5265.18.
*   As cidades com a maior pontuação, satisfação do cliente, são Naypyitaw e Yangon, com 7.07 e 7.02 respectivamente.
*   A proporção entre membros e clientes é de 1:1.
*   A linha produtos com maior lucro é alimenticio, com lucro de 2673.56.
*   O produto mais coomprado pelo gênero feminino é acessórios de moda, enquanto do gênero masculino são produtos de saúde e beleza.
*   A loja de Yangon é a que mais vende produtos de casa e estilo de vida.
*   Naypyitaw é a que mais vende produtos alimenticios.
*   Mandalay a que mais vende produtos esportivos e de viagem.
*   Os métodos de pagamento mais recorrentes são dinheiro e e-wallet, com a mesma frequência.
*   Em Yangon é mais comum e-wallet e em Naypyitaw dinheiro. Não há preferência em Mandalay.
*   O gênero feminino tende a pagar mais em dinheiro e o masculino com e-wallet.
*   A proporção de clientes do gênero masculino para o gênero feminino é de 1:1.

